### PR TITLE
Suggested fix for #1000: not passing Command Line Args to dotnet projects, when launching from Visual Studio's debugger

### PR DESCRIPTION
**Pull request by jaredthirsk**
For issue #1000 

As mentioned in the issue:
 - When **not debugging**, ApplicationExecutor.cs PrepareProjectExecutables() will construct the command line arguments from the launchProfile. 
 - It looks like this logic is either missing in Aspire **when debugging**, or dotnet.exe is supposed to be picking up the launch profile some other way (DOTNET_LAUNCH_PROFILE?) but it's not working for me.
 - This code works for my particular use case, but I'm not sure if there needs to be a "dotnet run" or "--" or whatnot to properly submit the args to the project.

### CODE REVIEW

File: src/Aspire.Hosting/Dcp/ApplicationExecutor.cs [hunk 1]
```diff
@@ -294,6 +294,21 @@ private void PrepareProjectExecutables()
...
+                    var launchProfile = project.GetEffectiveLaunchProfile();
+                    if (launchProfile is not null && !string.IsNullOrWhiteSpace(launchProfile.CommandLineArgs))
+                    {
+                        var cmdArgs = launchProfile.CommandLineArgs.Split((string?)null, StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries);
+                        if (cmdArgs is not null && cmdArgs.Length > 0)
+                        {
+                            exeSpec.Args = [];
+                            exeSpec.Args.AddRange(cmdArgs);
+                        }
+                    }
```
Comment (davidfowl) [inline src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:310]: This logic is duplicated, we should do it once and reuse it if it's already been done. (resolving the effective launch profile)

Comment (davidfowl) [reply src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:310]: I guess we also need to make sure that this doesn't break anything? It's not clear why this resolution is only in the else block.

@karolz-ms @mitchdenny

Comment (davidfowl) [reply src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:310]: I guess this is a branch so we can't share the resolution but we can put it in a method

**Comment by jaredthirsk** [comment]
@dotnet-policy-service agree

**Comment by karolz-ms** [comment]
Thank you @jaredthirsk  for submitting the issue and taking a stab at a fix--very much appreciated. 

I have added a comment to https://github.com/dotnet/aspire/issues/1000 explaining why I suspect this is Visual Studio issue and why I think this change is probably not what we want.

**Comment by jaredthirsk** [comment]
Issue #1000 closed instead by https://github.com/dotnet/aspire/pull/1212 (wiring up "csharp-launch-profile").

**Pull request closed without merging**